src/Elmish.WPF/BindingData.fs

```diff
@@ -647,11 +647,11 @@ module internal BindingData =
     let boxMinorTypes d = d |> mapMinorTypes box box box unbox unbox unbox
 
     let create createViewModel updateViewModel =
-      { GetModels = id
+      { GetModels = (fun x -> upcast x)
```

Comment:
This change seems like the opposite of the other changes.  In the other changes, you replaced a nontrivial function (namely `box`) with the trivial function `id`.  Here you are doing the opposite: you replaced the trivial function `id` with the nontrivial function `fun x -> upcast x`.

Why this change instead of mapping the model by this function in `Binding.fs`?

Same old/new line 702.

Just like we have names for functions like `Func2.id1` and `Func.id2`, I tried to extract `fun x -> upcast x` to a named function, but I was never able to make the F# complier happy.  Can you try to do this?

Reply:
It's making so that you can use `#seq<'a>` instead of `seq<'a>`. I think one of them was already doing it so I just added it to the other two list ones.

Reply:
Ok. Let's keep them, but move them to `Bindind.fs` just like `box`.  And if you can figure out how to extract it to a function, even better.

Reply:
I'm not sure it belongs there. The reason to move `box` out is because there are implementations that won't use it. However, this upcast is pretty fundamental to any given implementation of `BindingData`. Also this function is only manipulating the type system, it's not actually doing anything.

Reply:
Ok